x-pack/plugins/alerting/public/plugin.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { CoreSetup, Plugin, CoreStart } from 'src/core/public';
+
+import { AlertNavigationRegistry, AlertNavigationHandler } from './alert_navigation_registry';
+import { loadAlert, loadAlertType } from './alert_api';
+import { Alert, AlertNavigation } from '../common';
+
+export interface PluginSetupContract {
+  registerNavigation: (
+    consumer: string,
+    alertType: string,
+    handler: AlertNavigationHandler
+  ) => void;
+  registerDefaultNavigation: (consumer: string, handler: AlertNavigationHandler) => void;
+}
+export interface PluginStartContract {
+  getNavigation: (alertId: Alert['id']) => Promise<AlertNavigation | undefined>;
+}
+
+export class AlertingPublicPlugin implements Plugin<PluginSetupContract, PluginStartContract> {
+  private alertNavigationRegistry?: AlertNavigationRegistry;
+  public setup(core: CoreSetup) {
+    const alertNavigationRegistry = new AlertNavigationRegistry();
+    this.alertNavigationRegistry = alertNavigationRegistry;
+    return {
+      registerNavigation: async (
```

Comment:
The split param is to avoid having to use (!) everywhere... OOP is such a silly pattern for this. 😫 

TBH I'm not sure how to simplify this further, these are literally one liner functions.

Perhaps if you suggested something you'd feel more comfortable with?

Reply:
Looking at the current code, where `registerNavigation` and `registerDefaultNavigation`  functions are declared as variables, then the final return is the two variable references is **much** more readable to me, thx!

Reply:
👍